docs/user-documentation.rst

```diff
@@ -0,0 +1,425 @@
+Depending on the framework's architecture, invoking something at the end of a
+request may mean creating some sort of WSGI middleware, or implementing an
+event handler. For example, as WSGI middleware::
+
+    from webob import Request
+    from switchboard.signals import request_finished
+
+    class SwitchboardMiddleware(object):
+
+        def __init__(self, app):
+            self.app = app
+
+        def __call__(self, environ, start_response):
+            req = resp = None
+            try:
+                req = Request(environ)
+                resp = req.get_response(self.app)
+                return resp(environ, start_response)
+            finally:
+                self._end_request(req)
+
+        def _end_request(self, req):
+            if req:
+                # Notify Switchboard that the request is finished
+                request_finished.send(req)
+
+
+Using switches
+==============
+
+Once Switchboard is up and running within the application, it's time to begin
+using switches within the code. By default, Switchboard is set to autocreate
+switches, which means that a switch just needs to be checked in code and if
+it doesn't exist it will be created and disabled by default. A switch is always
+referred to by its key, a string name that is expected to be unique to that
+key.
+
+In Python
+---------
+
+To use in Python code (views, models, etc.), import the operator singleton
+and use the ``is_active`` method to see if the switch is on or not::
+
+    from switchboard import operator
+    ...
+    if operator.is_active('foo'):
+        ... do something ...
+    else:
+        ... do something else ...
+
+If autocreate is on (and it is by default), the 'foo' switch will be
+automatically created and set to disabled the first time it is referenced.
+Activating the switch and controlling exactly when the switch is active,
+are covered in `Managing switches`_.
+
+In templates
+------------
+
+Every templating framework has its own take on how (or even if) logic may be
+used. That said, Switchboard provides some helpers to make things easier. If
+
+In javascript
+-------------
+
+The easiest way to use Switchboard in conjunction with Javascript is to set a
+Javascript flag within your template code and then use that flag within your
+Javascript logic. Using Mako's syntax in the template::
+
+    <%!
+        from switchboard import operator
+    %>
+    <script>
+        window.switches = window.switches || {};
+        % if operator.is_active('foo'):
+        switches.foo = true;
+        % else:
+        switches.foo = false;
+        % endif
+    </script>
+
+In the Javascript::
+
+    if (switches.foo) {
+        ... do something ...
+    } else {
+        ... do something else ...
+    }
+
+Again, this time using Jinja syntax and the Switchboard-provided "active"
+test_::
+
+    <script>
+        window.switches = {};
+        switches.foo = {{ 'true' if 'foo' is active else 'false' }};
+    </script>
+
+Context objects
```

Comment:
Should `Custom conditions` come before `Context objects` ? I don't understand how it would take advantage of the new context data without custom conditions.